presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -558,7 +566,7 @@ protected static CacheQuota generateCacheQuota(HiveSplit hiveSplit)
                     tableToPartitionMapping.getPartitionSchemaDifference(),
                     tableToPartitionMapping.getTableToPartitionColumns());
             List<Column> partitionKeyColumns = partitionKeyColumnHandles.stream()
-                    .map(handle -> Column.partitionColumn(handle.getName(), handle.getHiveType(), handle.getComment()))
+                    .map(columnHandle -> columnHandle.toColumn(session, columnConverterProvider))
```

Comment:
If we look closer into how `partitionKeyColumns` is used, the type metadata is actually not used by the downstream. So we can get a list of names and a list of hive types to be piped into downstream. That should be enough. We don't need columnConverterProvider here in the reader.